### PR TITLE
[ROCm][CI] Fix Entrypoints Integration (Pooling) tests on TheRock image

### PR DESCRIPTION
**Pull request by mawong-amd**
## Purpose
`tests/entrypoints/pooling/scoring/test_cross_encoder_online_vision.py` has a number of [failures on TheRock image](https://buildkite.com/vllm/amd-ci/builds/13085/list?sid=01a0ae47-79de-4154-ad16-daf19cd84150&tab=output).

This was root-caused to a slight difference in codegen on Triton 3.8 vs Triton 3.7: namely, `chunked_prefill_paged_decode`'s `_fwd_kernel` has a slightly different FP32 accumulation order which leads to a 1-ULP BF16 divergence out of 32768 elements in layer 3 given identical inputs; this 1-ULP divergence is enough to snowball over the succeeding 20+ layers.
It is important to note that both codegens are equally correct: the problematic value sits 0.034 ULP from a BF16 rounding boundary, and both codegens' values have the same max error vs a FP64 reference.

Fundamentally, the issue is that this test's golden values were designed for CUDA in #33060. ROCm's attention backends cannot reasonably be expected to have exactly the same numerics as CUDA, and indeed there have been 3 rounds of tolerance relaxation (#41341, #46891, #53646) in an attempt to reconcile the slightly different (yet still correct) numerics of ROCm's attention backends vs CUDA's golden values.
In particular, FlexAttention differs from the other backends in that it accumulates `P = softmax(QK^T)` in `Q.dtype` (which can be BF16/FP16 rather than FP32 used on all other backends on ROCm/CUDA), so it naturally has more numerical divergence.
Rather than adding yet more attention-backend-specific tolerances, the proper fix is to establish a ROCm-specific set of golden values tuned for its attention backends, which also allows the use of a unified set of `atol` and `rtol` rather than the current fragmented regime.

## Test Plan
AMD CI (in particular the `Entrypoints Integration (Pooling)` test group) was run on both [ROCm 10](https://buildkite.com/vllm/amd-ci/builds/13103) and [ROCm 7.2](https://buildkite.com/vllm/amd-ci/builds/13108/list), on both MI300X and MI355X.

## Test Result
The test group now passes (on ROCm 10) and there are no new regressions (on ROCm 7.2).

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by rasmith** [comment]
@mawong-amd  It fails due to this Triton change: https://github.com/triton-lang/triton/pull/9724/changes. Also, all the golden values are stale and should probably be lazily computed just in case they do move (eg, there was a \n added to the prompts which made the golden values stale)

**Comment by mawong-amd** [comment]
> @mawong-amd It fails due to this Triton change: https://github.com/triton-lang/triton/pull/9724/changes

Based on the disassembly the accumulation is still done in FP32, just in a different order. So I wouldn't single out that change specifically as a regression. This test should be insensitive to FP32 accumulation order.

**Comment by rasmith** [comment]
> > @mawong-amd It fails due to this Triton change: https://github.com/triton-lang/triton/pull/9724/changes
> 
> Based on the disassembly the accumulation is still done in FP32, just in a different order. So I wouldn't single out that change specifically as a regression. This test should be insensitive to FP32 accumulation order.

If you build Triton 3.8 from source (I was using `0d6318389`) and do:
```
sed -i '310,316d' third_party/amd/lib/Dialect/TritonAMDGPU/IR/TargetFeatures.cpp \
sed -i '311d' third_party/amd/lib/Dialect/TritonAMDGPU/IR/TargetFeatures.cpp \
```
which effectively removes the optimization you'll get the same exact results as before.

I would like to bump the tolerance, but it does go from 2.5% error to almost 10% margin of error.  Not sure if @AndreasKaratzas  would accept this.

**Comment by rasmith** [comment]
> > @mawong-amd It fails due to this Triton change: https://github.com/triton-lang/triton/pull/9724/changes
> 
> Based on the disassembly the accumulation is still done in FP32, just in a different order. So I wouldn't single out that change specifically as a regression. This test should be insensitive to FP32 accumulation order.

Based on the testing I've done, it doesn't really look like a regression (I ran this through pytorch tests which compare against the CPU implementations and added vLLM tensors / dtypes and they all passed), but this model does fail this test with a larger margin of error.

**Comment by mawong-amd** [comment]
> > > @mawong-amd It fails due to this Triton change: https://github.com/triton-lang/triton/pull/9724/changes
> > 
> > 
> > Based on the disassembly the accumulation is still done in FP32, just in a different order. So I wouldn't single out that change specifically as a regression. This test should be insensitive to FP32 accumulation order.
> 
> If you build triton from source and do:
> 
> ```
> sed -i '310,316d' third_party/amd/lib/Dialect/TritonAMDGPU/IR/TargetFeatures.cpp \
> sed -i '311d' third_party/amd/lib/Dialect/TritonAMDGPU/IR/TargetFeatures.cpp \
> ```
> 
> which effectively removes the optimization you'll get the same exact results as before.
> 
> I would like to bump the tolerance, but it does go from 2.5% error to almost 10% margin of error. Not sure if @AndreasKaratzas would accept this.

Did you read/**understand** the PR body at all that says 1 ULP difference in BF16 in 1 element (and both roundings being equally accurate) is enough to cause the final divergence? Meaning this test as written currently is overly sensitive? What's the relevance of a margin of error to a test that's not a good measure of accuracy?

**Comment by rasmith** [comment]
> > > > @mawong-amd It fails due to this Triton change: https://github.com/triton-lang/triton/pull/9724/changes
> > > 
> > > 
> > > Based on the disassembly the accumulation is still done in FP32, just in a different order. So I wouldn't single out that change specifically as a regression. This test should be insensitive to FP32 accumulation order.
> > 
> > 
> > If you build triton from source and do:
> > ```
> > sed -i '310,316d' third_party/amd/lib/Dialect/TritonAMDGPU/IR/TargetFeatures.cpp \
> > sed -i '311d' third_party/amd/lib/Dialect/TritonAMDGPU/IR/TargetFeatures.cpp \
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > which effectively removes the optimization you'll get the same exact results as before.
> > I would like to bump the tolerance, but it does go from 2.5% error to almost 10% margin of error. Not sure if @AndreasKaratzas would accept this.
> 
> Did you read/**understand** the PR body at all that says 1 ULP difference in BF16 in 1 element (and both roundings being equally accurate) is enough to cause the final divergence? Meaning this test as written currently is overly sensitive? What's the relevance of a margin of error to a test that's not a good measure of accuracy?

Yes, I saw the same results, and I'm kinda OK with this change, I just wanted to add some more information. The goldens can be computed via CPU though (doesn't take too long).

**Review by AndreasKaratzas — APPROVED** [review]
